woocommerce/woocommerce
# Fix sticky admin bar CSS positioning bug in WordPress.com mobile

### Changes proposed in this Pull Request:





Closes https://github.com/Automattic/wp-calypso/issues/93333.

* This PR uses a slightly more specific CSS selector for `#wpadminbar` when styling for mobile.
* This fixes a CSS admin bar positioning bug in WordPress.com with WooCommerce on mobile.

#### Why?
WooCommerce uses the same CSS selector as core, simply #wpadminbar. It appears in most cases WP Core's CSS is processed first, followed by the WooCommerce plugin. However, for whatever reason (probably performance-related), WordPress.com processes WooCommerce's CSS first, so WP Core's CSS takes precedence because it's processed last. We could try to change the order in which these CSS files are processed on WordPress.com, but I think the more robust solution is to make the WooCommerce CSS selector slightly more targeted so it takes precedence regardless of the order in which the styles are processed.

Before | After
--|--
<img width="387" alt="Screenshot 2024-08-15 at 3 08 45 PM" src="https://github.com/user-attachments/assets/d788c9e4-df6c-4af9-a41b-89b67267ec20">  | <img width="387" alt="Screenshot 2024-08-15 at 3 11 57 PM" src="https://github.com/user-attachments/assets/97674218-29d4-4dee-8228-bf08e5266a56">



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check for regressions by using the [WordPress Playground test site linked below](https://github.com/woocommerce/woocommerce/pull/50709#issuecomment-2291636862)
2. While on the Playground test site, test WooCommerce pages like /wp-admin/admin.php?page=wc-orders on mobile and ensure the WordPress admin bar and WooCommerce header look and function correctly.
3. Test this PR by using the [WooCommerce Beta Tester plugin](https://woocommerce.com/products/woocommerce-beta-tester/)
4. With the beta tester plugin installed, go to /wp-admin/admin.php?page=wc-admin&path=%2Flive-branches and select and activate this PR as the "live branch"
5. Again, test WooCommerce pages like /wp-admin/admin.php?page=wc-orders on mobile and ensure the WordPress admin bar and WooCommerce header look and function correctly.
